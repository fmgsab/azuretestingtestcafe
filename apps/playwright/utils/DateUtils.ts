export function formatDate(date: Date) {
    const day = String(date.getDate()).padStart(2, '0');
    const month = String(date.getMonth() + 1).padStart(2, '0');
    const year = date.getFullYear();

    return `${day}/${month}/${year}`;
}

export function getYearFromDate(date: Date) {
    return date.getFullYear();
}

export function getDate(numberOfYearsToMoveForwardOrBackward: number) : string{
    let date =  new Date(new Date().setFullYear(new Date().getFullYear() + numberOfYearsToMoveForwardOrBackward));
    return formatDate(date)
}

export function getYear(numberOfYearsToMoveForwardOrBackward: number) : string{
    let date =  new Date(new Date().setFullYear(new Date().getFullYear() + numberOfYearsToMoveForwardOrBackward));
    return getYearFromDate(date).toString()
}
