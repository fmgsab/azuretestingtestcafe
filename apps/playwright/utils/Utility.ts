import { expect, Locator, Page, test, TestInfo } from '@playwright/test';
import fs from 'fs';
import * as fastcsv from 'fast-csv';

export async function delay(ms: number) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

export async function screenshot(
  testInfo: TestInfo,
  page: Page,
  screenshotName: string,
  locator: Locator = page
    .getByRole('navigation')
    .locator('div')
    .filter({ hasText: 'Skip to canvasStorybookSearch for components/ComponentsButtonsForm WidgetsGridsM' })
    .nth(3)
) {
  await testInfo.attach(screenshotName, {
    body: await page.screenshot({
      mask: [locator],
    }),
    contentType: 'image/png',
  });
}

export async function readCSVData(filePath: string): Promise<any[]> {
  return new Promise((resolve) => {
    const vehicleFieldsData: any[] = [];

    fs.createReadStream(filePath)
      .pipe(fastcsv.parse({ headers: true }))
      .on('data', (data: any) => {
        vehicleFieldsData.push(data);
      })
      .on('end', () => {
        resolve(vehicleFieldsData);
      });
  });
}

function getJSONAdminData(): any {
  const adminData = fs.readFileSync('../../packages/models/src/data-dictionary/lookup-data.json', 'utf-8');
  return JSON.parse(adminData);
}

export function getEngineTypeArray(vehicleType: string): any {
  let engineTypesArray = [];
  for (const item of getJSONAdminData().engineTypes) {
    if (item.itemSubtype.includes(vehicleType)) {
      engineTypesArray = item.list;
      break;
    }
  }
  if (engineTypesArray.length < 1) {
    fail('Test failed -------> No Engine Type found for vehicle ' + vehicleType);
  }
  return engineTypesArray;
}

/**
 * typeToExtract is the field name on lookup-data.json file**/

export function getListArray(itemTypeToMatch: string, typeToExtract: string): any {
  let listArray = [];
  const genericTypes = getJSONAdminData()[typeToExtract];
  if (itemTypeToMatch != '') {
    for (const item of genericTypes) {
      if (item.itemType.includes(itemTypeToMatch)) {
        listArray = item.list;
        break;
      }
    }
  } else {
    listArray = genericTypes;
  }
  console.log(listArray);
  return listArray;
}

export async function checkInputField(
  page: Page,
  frameLocator: string,
  fieldLocator: string,
  stateOfTheField: string,
  valueToPopulate: string
) {
  if (stateOfTheField.toUpperCase().includes('HIDE')) {
    await expect(page.frameLocator(frameLocator).getByTestId(fieldLocator)).toHaveCount(0);
  } else {
    await expect(page.frameLocator(frameLocator).getByTestId(fieldLocator)).toHaveCount(1);
    await page.frameLocator(frameLocator).getByTestId(fieldLocator).fill(valueToPopulate);
  }
}

export async function checkFieldByLocator(
  page: Page,
  frameLocator: string,
  fieldName: string,
  locatorFieldName: string,
  stateOfTheField: string,
  valueToPopulate: string,
  comboBox: Boolean = false
) {
  if (stateOfTheField.toUpperCase().includes('HIDE')) {
    await test.step('Verify the field ' + fieldName + ' is not displayed in the form', async () => {
      await expect(page.frameLocator(frameLocator).locator(locatorFieldName)).toHaveCount(0);
    });
  } else {
    await test.step('Verify the field ' + fieldName + ' is displayed in the form', async () => {
      await expect(page.frameLocator(frameLocator).locator(locatorFieldName)).toHaveCount(1);
      if (valueToPopulate != '') {
        if (!comboBox) {
          await page.frameLocator(frameLocator).locator(locatorFieldName).click();
          await page.frameLocator(frameLocator).locator(locatorFieldName).fill(valueToPopulate);
        } else {
          // await page.frameLocator(frameLocator).locator(locatorFieldName).fill(valueToPopulate);
          await page.frameLocator(frameLocator).locator(locatorFieldName).click();
          await page
            .frameLocator(frameLocator)
            .locator(locatorFieldName + '/input')
            .fill(valueToPopulate);
          await page.frameLocator(frameLocator).getByText(valueToPopulate, { exact: true }).click();
        }
      }
    });
  }
}

export async function checkTwoInputFields(
  page: Page,
  frameLocator: string,
  fieldName: string,
  locatorFieldName: string,
  locatorFieldName2: string,
  stateOfTheField: string,
  valueToPopulate: string
) {
  if (stateOfTheField.toUpperCase().includes('HIDE')) {
    await test.step('Verify the field ' + fieldName + ' is not displayed in the form', async () => {
      await expect(page.frameLocator(frameLocator).locator(locatorFieldName)).toHaveCount(0);
    });
  } else {
    await test.step('Verify the field ' + fieldName + ' is displayed in the form', async () => {
      await expect(page.frameLocator(frameLocator).locator(locatorFieldName)).toHaveCount(1);
      await page.frameLocator(frameLocator).locator(locatorFieldName).fill(valueToPopulate);
      await expect(page.frameLocator(frameLocator).locator(locatorFieldName2)).toHaveCount(1);
    });
  }
}

/**
 * fieldName is the exact field name from the form
 * locatorFieldName is the div id locator that comprises the radio button options **/
export async function checkRadioOrCheckBoxField(
  page: Page,
  frameLocator: string,
  fieldName: string,
  locatorFieldName: string,
  stateOfTheField: string,
  listArray: any,
  defaultValueToCheck: string,
  valueToSelect: string,
  checkBox: Boolean = false
) {
  if (stateOfTheField.toUpperCase() == 'HIDE') {
    await test.step('Verify the field ' + fieldName + ' is not displayed in the form', async () => {
      await expect(page.frameLocator(frameLocator).locator(locatorFieldName)).toHaveCount(0);
    });
  } else {
    await expect(page.frameLocator(frameLocator).locator(locatorFieldName)).toHaveCount(1);
    for (const list of listArray) {
      if (!checkBox) {
        await test.step('Verify the field ' + list + ' is displayed in ' + fieldName, async () => {
          await expect(page.frameLocator(frameLocator).locator(locatorFieldName).getByText(list, { exact: true })).toHaveCount(1);
        });
      } else {
        await test.step('Verify the field ' + list.label + ' is displayed in ' + fieldName, async () => {
          await expect(page.frameLocator(frameLocator).locator(locatorFieldName).getByText(list.label, { exact: true })).toHaveCount(1);
        });
      }
    }
    if (defaultValueToCheck != '') {
      await test.step('Verify the default ' + defaultValueToCheck + ' is selected in ' + fieldName, async () => {
        await expect(page.frameLocator(frameLocator).locator("css=input[value='" + defaultValueToCheck + "'][checked]")).toHaveCount(1);
      });
    }
    if (valueToSelect != '') {
      await page.frameLocator(frameLocator).locator(locatorFieldName).getByText(valueToSelect).click();
    }
  }
}

export async function checkRadioOrCheckBoxWithInputField(
  page: Page,
  frameLocator: string,
  fieldName: string,
  locatorFieldName: string,
  stateOfTheField: string,
  listArray: any,
  defaultValueToCheck: string,
  valueToSelect: string,
  checkBox: Boolean = false,
  inputFieldName: string,
  inputFieldLocator: string,
  inputValueToPopulate: string
) {
  await checkRadioOrCheckBoxField(
    page,
    frameLocator,
    fieldName,
    locatorFieldName,
    stateOfTheField,
    listArray,
    defaultValueToCheck,
    valueToSelect,
    checkBox
  );
  if (inputValueToPopulate != '') {
    // await page.frameLocator(frameLocator).locator(locatorFieldName).getByText(valueToSelect).click();
    // await page.frameLocator(frameLocator).locator(locatorFieldName).scrollIntoViewIfNeeded();
    await checkFieldByLocator(page, frameLocator, inputFieldName, inputFieldLocator, stateOfTheField, inputValueToPopulate);
  }
  // }
}
