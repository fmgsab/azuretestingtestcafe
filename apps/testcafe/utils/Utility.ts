export async function delay(ms: number) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

function getJSONAdminData(): any {
  const adminData = require('../../../packages/models/src/data-dictionary/lookup-data.json');
  return adminData;
}

export function getEngineTypeArray(vehicleType: string): any {
  let engineTypesArray = [];
  for (const item of getJSONAdminData().engineTypes) {
    if (item.itemSubtype.includes(vehicleType)) {
      engineTypesArray = item.list;
      break;
    }
  }
  if (engineTypesArray.length < 1) {
    fail('Test failed -------> No Engine Type found for vehicle ' + vehicleType);
  }
  return engineTypesArray;
}

/**
 * typeToExtract is the field name on lookup-data.json file**/

export function getListArray(itemTypeToMatch: string, typeToExtract: string): any {
  let listArray = [];
  const genericTypes = getJSONAdminData()[typeToExtract];
  if (itemTypeToMatch != '') {
    for (const item of genericTypes) {
      if (item.itemType.includes(itemTypeToMatch)) {
        listArray = item.list;
        break;
      }
    }
  } else {
    listArray = genericTypes;
  }
  console.log(listArray);
  return listArray;
}
