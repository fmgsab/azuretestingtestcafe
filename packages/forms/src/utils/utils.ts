const filterItems = (items: Record<string, string>[], lookup: string) => items.filter((item) => item?.itemType?.includes(lookup));
const getItemtypes = (data: object) => Object.values(data).flatMap((key) => key);

export const getAllItemTypes = (data: object, lookup: string) => {
  return filterItems(getItemtypes(data), lookup);
};
