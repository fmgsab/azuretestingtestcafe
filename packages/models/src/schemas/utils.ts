export function extractPathValue(args: Record<string, unknown>, path: string): unknown {
  const allPaths = path.split('.');
  if (allPaths.length > 1) {
    const currentPath = allPaths.at(0);
    const nextPath = allPaths.slice(1).join('.');
    return currentPath && args[currentPath] ? extractPathValue(args[currentPath] as Record<string, unknown>, nextPath) : '';
  }
  return args[path];
}
