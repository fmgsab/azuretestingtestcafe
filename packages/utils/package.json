{
  "name": "@fmg/utils",
  "version": "1.0.0",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "scripts": {
    "check-types": "tsc --noEmit --skipLibCheck",
    "lint": "eslint **/*.{tsx,ts,js,jsx}",
    "coverage": "vitest --coverage",
    "test": "vitest",
    "test:ui": "vitest --ui --coverage.enabled=true"
  },
  "dependencies": {
    "tsconfig": "workspace:*"
  }
}
