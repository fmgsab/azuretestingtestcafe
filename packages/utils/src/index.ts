export { camelize } from './string/camelize';
export { stringToKebab } from './string/kebab';
export * from './array/concat';
export * from './array/pick';
export * from './string/to-number';
export * from './string/to-formatted-date';
export * from './debounce/debounce';
export { alpha, alphaNumeric, alphaNumericSpaceless } from './regex/regex-patterns';
export { checkSizeIfSet } from './set/set';
