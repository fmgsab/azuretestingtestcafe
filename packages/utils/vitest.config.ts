import { configDefaults, defineConfig } from 'vitest/config';

export default defineConfig({
  plugins: [],

  test: {
    globals: true,
    environment: 'jsdom',
    css: false,
    coverage: {
      all: true,
      provider: 'v8',
      reporter: ['text', 'html', 'json'],
      reportOnFailure: true,
      include: ['**/src/**'],
      exclude: [
        ...configDefaults.exclude,
        '**/*.stories.*',
        '**/*.table.*',
        '**/*.test.*',
        '**/*/types',
        '**/*/test',
        '**/*index.ts',
        '**/*d.ts',
        '**/*types.ts',
        '**/*test.ts',
        '**/*schema.ts',
      ],
    },
    exclude: [...configDefaults.exclude],
  },
});
