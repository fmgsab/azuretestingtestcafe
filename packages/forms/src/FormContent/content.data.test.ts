import lookupData from 'models/src/data-dictionary/lookup-data.json';
import { describe } from 'vitest';
import { getAllItemTypes } from '../utils/utils';

describe('content lookup-data snapshot', () => {
  const itemTypes = getAllItemTypes(lookupData, 'content');
  it(`should match snapshot`, () => {
    expect(itemTypes).toMatchSnapshot();
  });
});
