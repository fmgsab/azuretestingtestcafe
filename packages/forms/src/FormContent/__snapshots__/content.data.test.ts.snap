// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`content lookup-data snapshot > should match snapshot 1`] = `
[
  {
    "itemType": [
      "content",
    ],
    "list": [
      "Household contents",
      "Landlord's contents",
      "Holiday home contents",
    ],
  },
  {
    "itemType": [
      "content",
    ],
    "list": [
      "Nominated Replacement Value",
      "Present Day Value",
    ],
  },
  {
    "itemSubtype": [
      "Household contents",
    ],
    "itemType": [
      "content",
    ],
    "list": [
      "Owner/Family Member",
      "Employee",
      "Renting",
      "Unoccupied",
    ],
  },
  {
    "itemType": [
      "content",
    ],
    "list": [
      "300",
      "500",
      "1,000",
      "2,000",
      "5,000",
      "10,000",
    ],
  },
  {
    "itemType": [
      "content",
    ],
    "list": [
      "Battery Powered Transportation",
      "Bicycles",
      "Collection",
      "Dentures",
      "Drones or Model Aircraft",
      "Hearing Aids",
      "Jewellery",
      "Sports and Recreational Equipment",
      "Works of Art",
      "Other High Value Items",
    ],
  },
]
`;
